core/src/client/mod.rs

```diff
@@ -110,10 +111,28 @@ pub trait SubscriptionClientT: ClientT {
 }
 
 /// Transport interface to send data asynchronous.
+#[cfg(target_arch = "wasm32")]
+#[async_trait(?Send)]
+/// Transport interface for an asynchronous client.
+pub trait TransportSenderT: 'static {
```

Comment:
So I think the only difference between these wasm and "normal" traits are that wasm doesn't require `Send`. What would be the issue if `Send` existed here?

I wonder whether some duplication could be avoided by eg having a trait like `MaybeSend` that either implies `Send` or does not based on the feature flag, and then use that here instead?

Reply:
This is `gloo-net` related those types are wrapped in `Rc` and other non-send types so won't work. If you have a look you can see that we split the `WebSocket type` to a sender and receiver which impls `TransportSenderT` and `TransportReceiverT`

First I added extra channel for this i.e, another to use in the `Sender` and `Receiver` types but I got so annoyed by the extra overhead.

Given our use-case i.e. `wasm-bindgen-futures` only can produce !Send futures it might be safe to implement `Send` for our types so I did this hack instead.

With that said `MaybeSend` would be really neat but not sure how to do it with `async-trait`

Reply:
I've no idea whether this would work, but I wondered about something like:

```
#[cfg(target_arch = "wasm32")]
trait MaybeSend: Send {}

#[cfg(not(target_arch = "wasm32"))]
trait MaybeSend {}

impl <T> MaybeSend for T {}
```

and then for the attr on the wasm async_trait:

```
#[cfg_attr(target_arch = "wasm32", async_trait(?Send))]
```